### PR TITLE
Simplifying Tensorflow API

### PR DESCRIPTION
**Pull request by rbharath**
Simplifies tensorflow models to remove some cruft (No need to use wrapper `dc.models.TensorflowModel`). Since this changes the API, I'd like to run the full benchmark suite as a regression before merging this change in. Need to add in the machinery to enable such a run.

Edit: Changes in this PR:

- Removes `TensorflowModel` as a wrapper. Simplifies the use of tensorflow models as a results. Updates tests and examples for consistency.
- Fixes serious bug in `TensorflowMultiTaskRegressor` where dropout was turned on at test time. #291 issue raised for adding more tests.
- Adds `seed` option to `TensorflowGraphModel` to allow for deterministic training of tensorflow models.
- Adds `examples/run_all_benchmarks.py` to move closer to an automated benchmarking suite.

### CODE REVIEW

File: examples/benchmark.py [hunk 1]
```diff
@@ -183,51 +183,74 @@ def benchmark_train_and_valid(base_dir,train_dataset,valid_dataset,tasks,
...
+    # Building tensorflow MultiTaskDNN model
     dropouts = hyper_parameters['dropouts']
     learning_rate = hyper_parameters['learning_rate']
     layer_sizes = hyper_parameters['layer_sizes']
     batch_size = hyper_parameters['batch_size']
     nb_epoch = hyper_parameters['nb_epoch']
 
-    tensorflow_model = dc.models.TensorflowMultiTaskClassifier(len(tasks),
-          n_features,  learning_rate=learning_rate, layer_sizes=layer_sizes, 
+    model_tf = dc.models.TensorflowMultiTaskClassifier(len(tasks),
```
Comment (miaecle) [inline examples/benchmark.py]: One question here is that if we just use the default weight_init_stddevs and bias_init_consts, it will be hard to define a more than one hidden layer structure. Should we add that it or just using one layer is good enough here? Actually I have already done that on my local branch, I can initiate another PR about it.

Comment (rbharath) [inline examples/benchmark.py]: Yes, good point. We should support multiple hidden layers. Let's save it for the next PR.

**Comment by rbharath** [comment]
Output from new `run_all_benchmarks.py` script (but only for Tox21):

```
0
tox21,train,tensorflow,0.983664335207
tox21,valid,tensorflow,0.756456812951
tox21,time_for_running,60.7451791763

0
tox21,train,tf_robust,0.979069095248
tox21,valid,tf_robust,0.737839727856
tox21,time_for_running,94.0613889694

0
tox21,train,logreg,0.665568025507
tox21,valid,logreg,0.597268826622
tox21,time_for_running,79.2564570904

0
tox21,train,graphconv,0.877051839137
tox21,valid,graphconv,0.803637956955
tox21,time_for_running,225.725852966
```

Some more work needs to be done to standardize this process, but this is a decent first pass. Going to merge this PR in once cleaned-up and tests pass again.

**Comment by rbharath** [comment]
CC @flee2 Could you try rerunning the models we've been working on now that this fix is merged in?

**Comment by rbharath** [comment]
Here's the full benchmark run with tox21 and muv. Will try running with more datasets later:

```
0
tox21,train,tensorflow,0.983667748326
tox21,valid,tensorflow,0.756475828439
tox21,time_for_running,61.3862650394

0
muv,train,tensorflow,0.965666641563
muv,valid,tensorflow,0.718311575822
muv,time_for_running,517.847835064

0
tox21,train,tf_robust,0.979091735442
tox21,valid,tf_robust,0.741062464217
tox21,time_for_running,95.9305889606

0
muv,train,tf_robust,0.985774091849
muv,valid,tf_robust,0.67214763649
muv,time_for_running,708.814903975

0
tox21,train,logreg,0.664966703646
tox21,valid,logreg,0.596598265391
tox21,time_for_running,81.795799017

0
muv,train,logreg,0.741564250945
muv,valid,logreg,0.670531973708
muv,time_for_running,666.504188061

0
tox21,train,graphconv,0.873470061959
tox21,valid,graphconv,0.797643093617
tox21,time_for_running,230.052562952

0
muv,train,graphconv,0.753236323854
muv,valid,graphconv,0.777733130968
muv,time_for_running,2887.4968009
```
